File: Python/baekjoon/단계별로 풀어보기/브루트 포스/1018_체스판 다시 칠하기.py
# https://www.acmicpc.net/problem/1018

n, m = map(int, input().split())
board = []
for _ in range(n):
    board.append(list(input()))
chessB = ['B' if i % 2 else 'W' for i in range(8)]
chessW = ['W' if i % 2 else 'B' for i in range(8)]
chess1 = [chessB if i % 2 else chessW for i in range(8)]
chess2 = [chessW if i % 2 else chessB for i in range(8)]
cnt = []
for i in range(n - 7):
    for j in range(m - 7):
        cnt1 = 0
        cnt2 = 0
        for r in range(i, i + 8):
            for c in range(j, j + 8):
                if board[r][c] != chess1[r - i][c - j]:
                    cnt1 += 1
                if board[r][c] != chess2[r - i][c - j]:
                    cnt2 += 1
        cnt.append(min(cnt1, cnt2))
print(min(cnt))

File: C#/Baekjoon/브루트 포스/1018_체스판 다시 칠하기.cs
//https://www.acmicpc.net/problem/1018

int[] s = Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
bool[,] b = new bool[s[0], s[1]];
for (int i = 0; i < s[0]; i++)
{
    string r = Console.ReadLine();
    for (int j = 0; j < s[1]; j++)
    {
        b[i, j] = r[j] == 'B';
    }
}

int m = 64;
for (int r = 0; r < s[0] - 7; r++) for (int c = 0; c < s[1] - 7; c++)
    {
        int n = 0;
        for (int i = r; i < r + 8; i++) for (int j = c; j < c + 8; j++)
            {
                if (b[i, j] ^ (i - r + j - c) % 2 == 0) n++;
            }
        m = Math.Min(m, Math.Min(n, 64 - n));
    }
Console.Write(m);